pkg/util/node/node.go

```diff
@@ -153,3 +154,27 @@ func SetNodeCondition(c clientset.Interface, node types.NodeName, condition v1.N
 	_, err = c.Core().Nodes().PatchStatus(string(node), patch)
 	return err
 }
+
+// UpdateNodeStatus updates node status with PatchStatus.
+func UpdateNodeStatus(c clientset.Interface, nodeName types.NodeName, oldNode *v1.Node, newNode *v1.Node) (*v1.Node, error) {
+	oldData, err := json.Marshal(oldNode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
+	}
+
+	newData, err := json.Marshal(newNode)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal new node %#v for node %q: %v", newNode, nodeName, err)
+	}
+
+	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create patch for node %q: %v", nodeName, err)
+	}
+
+	updatedNode, err := c.Core().Nodes().PatchStatus(string(nodeName), patchBytes)
```

Comment:
Could you switch to use the regular Patch() method? It supports specifying subresources now. We should deprecate the special PatchStatus() method.